PixaView/Controller/PixaViewController.swift

```diff
@@ -0,0 +1,276 @@
+    private func setupCollectionView(){
+        self.collectionView.register(UINib(nibName: PixaViewCell.cellIdentifier, bundle: nil), forCellWithReuseIdentifier: PixaViewCell.cellIdentifier) //assign to cell the Xib file
+        self.collectionView.dataSource = self //the cell need to understand where to get the filling from. add protocol to extension - UICollectionViewDataSource
+        self.collectionView.delegate = self //delegate with collectionView. Subscribe to UICollectionViewDelegate as delegate
+    }
+    
+    
+    //    func saveSortStatus(){
+    //        let filterVC = TableFilterVC()
+    //        //chekedItems = filterVC.selectedCategories
+    //        chekedItems.append(contentsOf: filterVC.selectedCategories)
+    //        print ("main vc contains \(chekedItems)")
+    //    }
+    
+    //MARK: - fetch data
+    func fetch(order: Order, currentPage: Int, filterCategory: [Categories]){
+        let URLSession = URLSession.shared
+        let service: HTTPService = HTTPService(with: URLSessionHttpClient(session: URLSession))
+        //            service.fetchPics(order: order, filterCategory: filterCategory, currentPage: currentPage) { fetchedHits, total in
+        service.fetchPics(order: order, filterCategory: filterCategory, currentPage: currentPage) {result in
+            
+            switch result{
+            case .success(( let hitsRESULT, let total )):
+                DispatchQueue.main.async {
+                    self.total = total
+                    self.hitsRESULT.append(contentsOf: hitsRESULT)
+                    self.collectionView.reloadData()
+                } // dispatch
+                
+            case .failure(let error):
+                DispatchQueue.main.async {
+                    let error = error.localizedDescription
+                    let alert = UIAlertController(title: "Алярм", message: error, preferredStyle: .alert)
+                    let okButton = UIAlertAction(title: "понял принял", style: .default)
+                    let  retry = UIAlertAction.init(title: "понял принял", style: .default) { _ in
+                        self.fetch(order: order, currentPage: currentPage, filterCategory: filterCategory)
+                    }
+                    //alert.addAction(okButton)
+                    alert.addAction(retry)
+                    self.present(alert, animated: true, completion: nil)
+                    
+                }
+            } // end of fetchPics
+        }
+    }
+    
+    //MARK: - End of <Methods>
+} // end of PixaViewController class
+
+//MARK: Extensions -
+extension PixaViewController:UICollectionViewDataSource, UICollectionViewDelegate, UICollectionViewDelegateFlowLayout, TableFilterViewControllerDelegate, SortImageProtocol {
+    
+    
+    //MARK: delegates for filter / sort -
+    //filter delegate
+    func filterButtonDidTap(filterCategory: [Categories]) {
+        hitsRESULT.removeAll()
+        currentPage = 1
```

Comment:
You could move these lines to dedicated method and use it inside filterButtonDidTap, sortInTableDidTap

```
        hitsRESULT.removeAll()
        currentPage = 1
```

```
func reset() {
       hitsRESULT.removeAll()
        currentPage = 1
}

  func sortInTableDidTap(sortButtodDidTap sort: Order) {
        reset()
        fetch(order: sort, currentPage: currentPage, filterCategory: category)
        order = sort
        print(sort)
    }
```

Reply:
thanks